src/goto-instrument/code_contracts.cpp

```diff
@@ -237,9 +251,51 @@ bool code_contractst::apply_contract(
     ++target;
   }
 
-  // overwrite the function call
-  *target = goto_programt::make_assumption(ensures, target->source_location);
+  // Create a series of non-deterministic assignments to havoc the variables
+  // in the assigns clause.
+  if(assigns.is_not_nil())
```

Comment:
This method is already way too big. Could you please improve the situation a bit by putting this branch in a method of its own?